Pull for diplom

L1gabas opened the pull request:
Добавлены тесты для классов Bun, Burger, Ingredient,IngredientType. использованы моки, стабы и параметризация там, где нужно.

L1gabas committed:
Добавлены тесты для классов Bun, Burger, Ingredient,IngredientType. Использованы моки, стабы и параметризация там, где нужно.

pomegranate282922 commented on src/test/java/BunTest.java:
```diff
@@ -0,0 +1,19 @@
+
+    @Test
+    public void priceReturnPrice() {
+        assertEquals(200, bun.getPrice(), 10);
```
Почему такая дельта?
Нужно исправить

pomegranate282922 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,90 @@
+    @Test
+    public void addIngredientTOBurger() {
+        burger.addIngredient(thousandIslandsSauce);
+        assertEquals(true, burger.ingredients.contains(thousandIslandsSauce));
```
assertTrue

pomegranate282922 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,90 @@
+        String burgerRecipe = burger.getReceipt();
+        assertEquals(burgerRecipe.trim(),burger.getReceipt().trim());
+
+    }
```
Ждем моков для всех зависимостей класса бургер 
Нужно исправить

pomegranate282922 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,90 @@
+
+    private Burger burger;
+
+    @Mock
```
Отлично что применен мок

pomegranate282922 reviewed (commented):
Работа хорошая! Применен мок. 
Замечания:
1. Поправить дельту 
2. Нет параметризации 
3. Нет тестов на enum 
4. Нужно применить моки для всех зависимостей класса бургер

L1gabas committed:
Добавлена параметризация, исправил дельту. Моки применены ко всем зависимостям класса бургер.
Тестов enum нет, так как на вебинаре обсуждали ненадобность их, так же в задании не описано, о их необходимости

pomegranate282922 commented on src/test/java/BunParamTest.java:
```diff
@@ -0,0 +1,38 @@
+
+    @Test
+    public void priceReturnPrice() {
+        assertEquals(200, bun.getPrice(), 0.5);
```
Чем объясняется такая большая дельта?
Нужно исправить

pomegranate282922 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,90 @@
+    public void setUp() {
+        burger = new Burger();
+        bunBlue = new Bun("Blue bun", 250);
+        thousandIslandsSauce = new Ingredient(IngredientType.SAUCE, "ThousandIslands", 120);
```
В чем смысл мока если здесь снова создаются реальные объекты?
Нужно исправить

pomegranate282922 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,90 @@
+        burger.addIngredient(fireAssSauce);
+        burger.addIngredient(fillingTomato);
+
+        assertEquals(1190, burger.getPrice(), 200);
```
Дельта?

pomegranate282922 commented on src/test/java/IngredientTest.java:
```diff
@@ -0,0 +1,22 @@
+    }
+    @Test
+    public void sauceReturnSaucePrice() {
+        assertEquals(100, bbq.getPrice(), 10);
```
Дельту?

pomegranate282922 reviewed (commented):
Многое исправил, но остались критичные замечания.

L1gabas committed:
Исправлена дельта, исправил моки

pomegranate282922 reviewed (approved):
Ок, зачет!
